lib/instrumentation/modules/fastify.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const semver = require('semver')
+
+module.exports = function (fastify, agent, version, enabled) {
+  if (!enabled) return fastify
+
+  agent.logger.debug('wrapping fastify build function')
+
+  return semver.gte(version, '2.0.0-rc') ? wrappedBuild2Plus : wrappedBuildPre2
```

Comment:
Pay attention that `semver` does not handle `-rc`, sometimes this can cause some unexpected behaviors.

Reply:
True, pre-release versions can be a little hard to reason about when it comes to semver. But in this case it should be ok. Normally I wouldn't measure against a pre-release version, but to get this working with the current pre-release version I had to do it this way.

```js
semver.gte('1.0.0', '2.0.0-rc') // => false
semver.gte('1.0.0-rc.0', '2.0.0-rc') // => false
semver.gte('2.0.0-rc.0', '2.0.0-rc') // => true
semver.gte('2.0.0', '2.0.0-rc') // => true
semver.gte('3.0.0', '2.0.0-rc') // => true
```

Reply:
You can remove the `rc` if you want 🚀